sources/database/DbTable-mysql.php

```diff
@@ -27,128 +27,84 @@ class DbTable_MySQL extends DbTable
-			'board_permissions', 'boards', 'calendar', 'calendar_holidays', 'categories', 'collapsed_categories',
-			'custom_fields', 'group_moderators', 'log_actions', 'log_activity', 'log_banned', 'log_boards',
-			'log_digest', 'log_errors', 'log_floodcontrol', 'log_group_requests', 'log_karma', 'log_mark_read',
-			'log_notify', 'log_online', 'log_packages', 'log_polls', 'log_reported', 'log_reported_comments',
-			'log_scheduled_tasks', 'log_search_messages', 'log_search_results', 'log_search_subjects',
-			'log_search_topics', 'log_topics', 'mail_queue', 'membergroups', 'members', 'message_icons',
-			'messages', 'moderators', 'package_servers', 'permission_profiles', 'permissions', 'personal_messages',
-			'pm_recipients', 'poll_choices', 'polls', 'scheduled_tasks', 'sessions', 'settings', 'smileys',
-			'themes', 'topics');
-
-		foreach ($this->_reservedTables as $k => $table_name)
-			$this->_reservedTables[$k] = strtolower($db_prefix . $table_name);
-
-		// let's be sure.
-		$this->_package_log = array();
-
-		// This executes queries and things
-		$this->_db = $db;
+		return preg_match('~^(`?)(.+?)\\1\\.(.*?)$~', $this->_db_prefix, $match) === 1 ? $match[3] : $this->_db_prefix;
 	}
 
 	/**
 	 * {@inheritdoc }
 	 */
-	public function db_create_table($table_name, $columns, $indexes = array(), $parameters = array(), $if_exists = 'ignore', $error = 'fatal')
+	protected function _close_table_query($temporary)
 	{
-		global $db_prefix;
 
-		// Strip out the table name, we might not need it in some cases
-		$real_prefix = preg_match('~^(`?)(.+?)\\1\\.(.*?)$~', $db_prefix, $match) === 1 ? $match[3] : $db_prefix;
+		$close_string = ') DEFAULT CHARSET=utf8 COLLATE=utf8_general_ci';
 
-		// With or without the database name, the fullname looks like this.
-		$full_table_name = str_replace('{db_prefix}', $real_prefix, $table_name);
-		$table_name = str_replace('{db_prefix}', $db_prefix, $table_name);
-
-		// First - no way do we touch our tables.
-		if (in_array(strtolower($table_name), $this->_reservedTables))
-			return false;
-
-		// Log that we'll want to remove this on uninstall.
-		$this->_package_log[] = array('remove_table', $table_name);
-
-		// Slightly easier on MySQL than the others...
-		if ($this->table_exists($full_table_name))
+		if ($temporary === true)
 		{
-			// This is a sad day... drop the table? If not, return false (error) by default.
-			if ($if_exists == 'overwrite')
-				$this->db_drop_table($table_name);
-			else
-				return $if_exists == 'ignore';
+			$close_string .= ' ENGINE=MEMORY';
 		}
 
-		// Righty - let's do the damn thing!
-		$table_query = 'CREATE TABLE ' . $table_name . "\n" . '(';
-		foreach ($columns as $column)
-			$table_query .= "\n\t" . $this->_db_create_query_column($column) . ',';
+		return $close_string;
+	}
 
+	/**
+	 * {@inheritdoc }
+	 */
+	protected function _create_query_indexes($indexes, $table_name)
```

Comment:
Best make $table_name optional... or update the calls.  https://scrutinizer-ci.com/g/elkarte/Elkarte/inspections/a0b14eb0-c2da-4421-b6e7-f12c21a93d76/issues/files/sources/database/DbTable-mysql.php?status=new&orderField=path&order=asc&honorSelectedPaths=0

Reply:
Done, along with few other tweaks. :innocent: